src/query/crypt_opfunc.c

```diff
@@ -305,83 +326,76 @@ crypt_aes_default_encrypt (THREAD_ENTRY * thread_p, const char *src, int src_len
   *dest_p = NULL;
   *dest_len_p = 0;
 
-  if (init_gcrypt () != NO_ERROR)
+  // *INDENT-OFF*
+  deleted_unique_ptr<EVP_CIPHER_CTX> context (EVP_CIPHER_CTX_new (), [] (EVP_CIPHER_CTX *ctxt_ptr)
+    {
+      if (ctxt_ptr != NULL)
+	{
+	  EVP_CIPHER_CTX_free (ctxt_ptr);
+	}
+    });
+  // *INDENT-ON*
+
+  if (context == NULL)
     {
+      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_ENCRYPTION_LIB_FAILED, 1,
+	      crypt_lib_fail_info[CRYPT_LIB_INIT_ERR]);
       return ER_ENCRYPTION_LIB_FAILED;
     }
 
-  i_gcrypt_err = gcry_cipher_open (&aes_ctx, GCRY_CIPHER_AES, GCRY_CIPHER_MODE_ECB, 0);
-  if (i_gcrypt_err != GPG_ERR_NO_ERROR)
+  if (EVP_EncryptInit (context.get (), cipher, (const unsigned char *) key_arg, NULL) != 1)
     {
       er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_ENCRYPTION_LIB_FAILED, 1,
-	      crypt_lib_fail_info[CRYPT_LIB_OPEN_CIPHER_ERR]);
+	      crypt_lib_fail_info[CRYPT_LIB_INIT_ERR]);
       return ER_ENCRYPTION_LIB_FAILED;
     }
 
   /* PKCS7 */
-  if ((src_len % AES128_BLOCK_LEN) == 0)
+  if ((src_len % block_len) == 0)
     {
-      pad = AES128_BLOCK_LEN;
+      pad = block_len;
       padding_src_len = src_len + pad;
     }
   else
     {
-      padding_src_len = (int) ceil ((double) src_len / AES128_BLOCK_LEN) * AES128_BLOCK_LEN;
+      padding_src_len = (int) ceil ((double) src_len / block_len) * block_len;
       pad = padding_src_len - src_len;
     }
 
   padding_src = (char *) db_private_alloc (thread_p, padding_src_len);
   if (padding_src == NULL)
     {
-      error_status = ER_OUT_OF_VIRTUAL_MEMORY;
-      goto exit_and_free;
+      return ER_OUT_OF_VIRTUAL_MEMORY;
     }
   memcpy (padding_src, src, src_len);
   memset (padding_src + src_len, pad, pad);
 
-  aes_default_gen_key (key, key_len, new_key, AES128_KEY_LEN);
-  i_gcrypt_err = gcry_cipher_setkey (aes_ctx, new_key, AES128_KEY_LEN);
-  if (i_gcrypt_err != GPG_ERR_NO_ERROR)
-    {
-      error_status = ER_ENCRYPTION_LIB_FAILED;
-      er_set (ER_ERROR_SEVERITY, ARG_FILE_LINE, ER_ENCRYPTION_LIB_FAILED, 1,
-	      crypt_lib_fail_info[CRYPT_LIB_SET_KEY_ERR]);
-      goto exit_and_free;
-    }
-
   dest = (char *) db_private_alloc (thread_p, padding_src_len);
   if (dest == NULL)
     {
-      error_status = ER_OUT_OF_VIRTUAL_MEMORY;
-      goto exit_and_free;
+      db_private_free_and_init (thread_p, padding_src);
+      return ER_OUT_OF_VIRTUAL_MEMORY;
     }
 
-  i_gcrypt_err = gcry_cipher_encrypt (aes_ctx, dest, padding_src_len, padding_src, padding_src_len);
-  if (i_gcrypt_err != GPG_ERR_NO_ERROR)
+  ciphertext_len = padding_src_len;
```

Comment:
Since '**ciphertext_len**' is a variable to return the number of bytes actually written in the EVP_EncryptUpdate function, this assignment seems to be meaningless.

Reply:
Removed it.